src/wp-admin/authorize-application.php

```diff
@@ -0,0 +1,187 @@
+		<?php if ( $app_name ) : ?>
+			<p>
+			<?php
+			/* translators: Application Name */
+			printf( __( 'Would you like to give the application identifying itself as %s access to your account?  You should only do this if you trust the app in question.' ), '<strong>' . esc_html( $app_name ) . '</strong>' );
+			?>
+			</p>
+		<?php else : ?>
+			<p><?php _e( 'Would you like to give this application access to your account? You should only do this if you trust the app in question.' ); ?></p>
+		<?php endif; ?>
+
+		<?php if ( $new_password ) : ?>
+			<p class="password-display">
+				<?php
+				printf(
+					/* translators: 1: Application name, 2: Generated password. */
+					__( 'Your new password for %1$s is %2$s.' ),
+					'<strong>' . esc_html( $app_name ) . '</strong>',
+					'<kbd>' . esc_html( WP_Application_Passwords::chunk_password( $new_password ) ) . '</kbd>'
+				);
+				?>
+			</p>
+		<?php else : ?>
+			<form action="<?php echo esc_url( admin_url( 'authorize-application.php' ) ); ?>" method="post">
+				<?php wp_nonce_field( 'authorize_application_password' ); ?>
+				<input type="hidden" name="action" value="authorize_application_password" />
+				<input type="hidden" name="success_url" value="<?php echo esc_url( $success_url ); ?>" />
+				<input type="hidden" name="reject_url" value="<?php echo esc_url( $reject_url ); ?>" />
+
+				<label for="app_name"><?php esc_html_e( 'Application Title:' ); ?></label>
+				<input type="text" id="app_name" name="app_name" value="<?php echo esc_attr( $app_name ); ?>" placeholder="<?php esc_attr_e( 'Name this connection&hellip;' ); ?>" required />
+
+				<p><?php submit_button( __( 'Yes, I approve of this connection.' ), 'primary', 'approve', false ); ?>
+					<br /><em>
+					<?php
+					if ( $success_url ) {
+						printf(
+							/* translators: The URL the user is being redirected to. */
+							__( 'You will be sent to %s' ),
+							'<strong><kbd>' . esc_html(
+								add_query_arg(
+									array(
+										'username' => $user->user_login,
+										'password' => '[------]',
+									),
+									$success_url
+								)
+							) . '</kbd></strong>'
+						);
+					} else {
+						_e( 'You will be given a password to manually enter into the application in question.' );
+					}
+					?>
+					</em>
+				</p>
+
+				<p><?php submit_button( __( 'No, I do not approve of this connection.' ), 'secondary', 'reject', false ); ?>
+					<br /><em>
+					<?php
+					if ( $reject_url ) {
+						printf(
+							/* translators: The URL the user is being redirected to. */
+							__( 'You will be sent to %s' ),
+							'<strong><kbd>' . esc_html(
+								add_query_arg(
+									array(
+										'success' => 'false',
+									),
+									$reject_url
+								)
+							) . '</kbd></strong>'
+						);
+					} else {
+						_e( 'You will be returned to the WordPress Dashboard, and we will never speak of this again.' );
```

Comment:
I understand the sentiment behind this, but I think there is a slight chance of someone interpreting this this is a permanent rejection for the application, and that it cannot be retried again later.

Reply:
YES you're absolutely right.  This was a somewhat playful tone I'd taken initially and as it's somewhat edge case (contingent on neither success or reject url being passed) it's never gotten reevaluated.

Totally should be revised to something a bit more in keeping with WordPress copy than what it is.

Reply:
Updated via 9a9ff7356496888c48644b7e16be69ab603e325d

Reply:
Updated in 9a9ff73.